.circleci/config.yml

```diff
@@ -29,11 +29,11 @@ templates:
           # If incremental dep fails, increase the cache gen number
           # in restore_deps AND save_deps
           # See https://github.com/DataDog/datadog-agent/pull/2384
-          - gen19-godeps-{{ .Branch }}-{{ .Revision }}
-          - gen19-godeps-{{ .Branch }}-
-          - gen19-godeps-main-
+          - gen19-godeps-{{ checksum "requirements.txt" }}-{{ checksum ".circleci/requirements.txt" }}-{{ .Branch }}-{{ .Revision }}
```

Comment:
It would be safer to do the checksum of `pip freeze` output (since `python3 -m pip install -r requirements.txt` may install different versions for unpinned indirect dependencies) but I don't know how hard that would be.

Reply:
I don't think this is possible. AFAICS we only have one dependency that is not "frozen" - `dulwich` in `requirements.txt`. So if we freeze that as well, we will actually have what you propose. Does that sound reasonable?

Reply:
Oh wait, scratch that. We have what appears to be frozen dependencies, but we don't freeze recursive dependencies, so freezing dulwich wouldn't help.

Reply:
Actually, it *is* possible: https://circleci.com/docs/2.0/caching/#creating-and-building-a-concatenated-package-lock-file - let me update the PR.

Reply:
>  but we don't freeze recursive dependencies, so freezing dulwich wouldn't help.

yes, that's what I meant :) we pin (more or less) all of our dependencies, but these dependencies may have unpinned dependencies themselves

Reply:
Hmm, on the second look, it doesn't seem to apply here, because while we can build a cache key from installed dependencies, we don't have it *before* we install them. So we could construct the key for saving cache, but not for retrieving it (without an actual pip freeze file). I think this is the best we can do 🤷

Reply:
Okay, let's go with this, I guess we can't do much if we don't use something with a lock file like `pipenv`, we can consider that in the future if we have problems with dependencies of dependencies